src/services/jsDoc.ts

```diff
@@ -142,33 +153,35 @@ namespace ts.JsDoc {
     }
 
     function getCommentDisplayParts(tag: JSDocTag, checker?: TypeChecker): SymbolDisplayPart[] | undefined {
-        const { comment } = tag;
-        switch (tag.kind) {
+        const { comment, kind } = tag;

```

Comment:
```suggestion
        const { comment, kind } = tag;
        const namePart = tagNameDisplayPart[kind];
        Debug.assertDefined(namePart, "No display part found for name of jsdoc comment: " + SyntaxKind[kind]);
```

and then you don't need to pass `syntaxKindDisplayPartFunctionDict` everywhere below

Reply:
Thanks. I gave these suggestions a go but it's causing lots of typing errors.
The `SyntaxKind` also contains lots of kinds that are not related to JSDoc so `Debug.assertDefined` causes lots of tests to fail.

I took a bit of a different approach to make the code cleaner. Let me know what you think.

Reply:
Interesting. I thought syntaxKindDisplayPartFunctionDict was exhaustive. Regardless, I like the current change.